pkg/rbac/auth.go

```diff
@@ -0,0 +1,87 @@
+package rbac
+
+import (
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/hobbyfarm/gargantua/v3/pkg/microservices"
+	"github.com/hobbyfarm/gargantua/v3/protos/authn"
+	"github.com/hobbyfarm/gargantua/v3/protos/authr"
+	userProto "github.com/hobbyfarm/gargantua/v3/protos/user"
+)
+
+func AuthenticateRequest(r *http.Request, caCertPath string) (*userProto.User, error) {
+	authnConn, err := microservices.EstablishConnection(microservices.AuthN, caCertPath)
```

Comment:
We should not be establishing a new grpc connection for every request that comes in. This is a lot of overhead - each time an API request is being authenticated, this will result in an entirely new TCP connection to be made, incl. tls negotiation, etc. 

Instead we should have long-lived grpc clients with retry/reconnection logic if the underlying stream is closed.